lib/fibonacci.rb

```diff
@@ -1,8 +1,22 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - should be O(n)
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
This works, but you are storing all the numbers in the Fibonacci sequence from 1 to n.  However you only need to keep track of the last two numbers in the sequence.  So you are using more memory than you need to.